torch/_dynamo/variables/torch_function.py

```diff
@@ -0,0 +1,195 @@
+from torch.overrides import _get_overloaded_args, get_default_nowrap_functions
+from torch.utils._pytree import tree_flatten
+from ..exc import unimplemented
+from ..source import AttrSource
+from .base import VariableTracker
+from .constant import ConstantVariable
+from .lists import TupleVariable
+from .user_defined import UserDefinedClassVariable
+
+# [Note: __torch_function__] This feature is partially supported with many rough edges (contact mlazos with issues):
+# At a high level, a torch function tensor subclass is represented as a TensorWithTFOverrideVariable, which dispatches
+# __torch_function__ on attribute accesses, method calls, and torch API calls.
+# The following is not supported:
+# - triggering __torch_function__ on tensor subclass non-tensor custom attributes
+# - graph breaking on mutating guardable tensor properties within a __torch_function__ context, this can cause
+# excessive recompiles in certain degenerate cases
+# - Matching the exact eager behavior of *ignoring* __torch_function__ objects in non-tensor argument positions of Torch API calls
+
+# The following is supported:
+# - static method impls of __torch_function__ on custom objects; this will trigger on torch API calls with the object as
+# any argument
+# - triggering __torch_function__ on torch API calls with tensor subclass arguments
+# - __torch_function__ calls on base tensor attribute access and method calls for tensor subclass instances
+# - matches the dispatch ordering behavior of eager __torch_function__ with subclass/object argumnents in any argument position
+
+# See https://docs.google.com/document/d/1WBxBSvW3NXhRp9ncmtokJloMLCtF4AYNhJaffvHe8Kw/edit#heading=h.vacn73lozd9w
+# for more information on the design.
+
+# To enable subclass behavior, add your tensor subclass type to traceable_tensor_subclasses in dynamo/config.py
+
+
+banned_attrs = [fn.__name__ for fn in get_default_nowrap_functions()]
```

Comment:
It is suspicious why you have to pull out the `__name__`. At minimum, this is relying on `get_default_nowrap_functions` only ever having a meaningful name, which is not implied at the def site (in particular, someone could conceivably add a `torch.blah` to the nowrap function list, which would then get mishandled here)

Reply:
I will extract the values that are attrs on tensor then, I was taking advantage of this fact implicitly (since they were all currently attrs on tensor). Basically I need to extract a string name to ban when accessing attrs.

Reply:
Yes, my point is that you are assuming that everything in `get_default_nowrap_functions` is an attribute accessor, and there is no intrinsic reason this necessarily has to be true

Reply:
I added a filter that only filters the values that are attrs